api/types.go

```diff
@@ -768,12 +768,15 @@ func (n *NodeWithMeta) Size() uint64 {
 // Metric transports information about a peer.ID. It is used to decide
 // pin allocations by a PinAllocator. IPFS cluster is agnostic to
 // the Value, which should be interpreted by the PinAllocator.
+// The TS value is a timestamp representing when a peer has received
+// the metric value.
 type Metric struct {
 	Name   string  `json:"name" codec:"n,omitempty"`
 	Peer   peer.ID `json:"peer" codec:"p,omitempty"`
 	Value  string  `json:"value" codec:"v,omitempty"`
 	Expire int64   `json:"expire" codec:"e,omitempty"`
 	Valid  bool    `json:"valid" codec:"d,omitempty"`
+	TS     int64   `json:"ts" codec:"t,omitempty"` // TS contains a UnixNano timestamp
```

Comment:
Seems you are making a distribution using the times metrics were received. Wouldn't it be better (or the same) to use the deltas between the time they were sent (and for that you can use the `Expire` field)?

Alternatively, we should name this `ReceivedAt` or something more self-explanatory.

Reply:
It needs to be between the received times as then it takes into account fluctuations in network latencies. There technical isn't going to be any difference between the expired value unless the peer disappears. 

Agreed with changing it to `ReceivedAt`.